### PR TITLE
[Feature] Restrict pools available in search to IT publishing group

### LINKED ISSUE #4373: ✨Restrict which pools are available for searching on the Search page
## ✨ Feature

Restrict the search page to counting candidates from `IT_JOBS` and `IT_JOBS_ONGOING` publishing groups. 

## 🕵️ Details

We don't want IAP candidates and exec candidates (in the future) to be included in the counts on the search talent page.

## 🙋‍♀️ Proposed Implementation

The `countApplicants` and `countPoolCandidatesByPool` queries should have a hard-coded scope applied to them which limits the results to candidates in `IT_JOBS` and `IT_JOBS_ONGOING` pools.

The `PoolCandidateSearchInput` input should have a new field added, which accepts an array of publishing group enums, and scopes to the results to candidates in those pools. A multiselect for this field should be added to the PoolCandidatesTable filter - and it should always default to `IT_JOBS` and `IT_JOBS_ONGOING`, so that initial results on a search request page match what the requester saw.

## ✅ Acceptance Criteria

- [ ] Talent Search counts on the client side are restricted to candidates to pools in the IT_JOBS and IT_JOBS_ONGOING publishing groups.
- [ ] It's impossible for non-admins to get candidate counts (via api) for other publishing groups. 
- [ ] Admins can filter PoolCandidates by any publishing group on the PoolCandidates table. The default value of this filter matches the search request.
- [ ] PHPUnit tests

### PR DESCRIPTION
**Pull request by brindasasi**
🤖 Resolves #4373

## 👋 Introduction

Updates the pool candidate searches to restirct candidates and the counts by IT publishing groups. Also, updates the pool factory to add a state to ensure the candidate's within the pool can appear in search results.

## 🕵️ Details

I'm a little unsure about the naming of these, any suggestions to better describe the purpose of  a few mehtods would be welcome 😄 

- `User::scopeIsIT`
- `Pool::factory()->candidatesAvailableInSearch()`

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Build `npm run dev`
2. Login as an admin
3. Navigate to search `/search`
4. Confirm it works as epexted (only counting candidates for IT pools)
5. Navigate to the pool candidate table
6. Confirm publishing group appears in filters and is set to both IR publishing groups by default
7. Confirm new publishing group filter works as expected

## 📸 Screenshot

Screenshot 2023-08-01 164442

### CODE REVIEW

**Comment by brindasasi** [comment]
@tristan-orourke I've updated the PR for the remaining task. Could you take over it ?

**Comment by vd1992** [comment]
> I'm a little unsure about the naming of these, any suggestions to better describe the purpose of a few mehtods would be welcome 😄
> 
>     * `User::scopeIsIT`
> 
>     * `Pool::factory()->candidatesAvailableInSearch()`

Hm...
`scopeInITPublishingGroup` maybe so that you can tell the purpose of the scope no matter where it is called from and what IT is referring to. 

`publishingGroupIT` suggestion for the latter as it keeps it simple as referring to other models and functions acting on other models, in this case searching on PoolCandidate, can make things a little confusing when calling it elsewhere. 
(a little tricky to name as "poolIT" doesn't read well and "iTPublishingGroup" look wrong)

File: apps/web/src/components/PoolCandidatesTable/PoolCandidatesTable.tsx [hunk 1]
```diff
@@ -340,6 +343,7 @@ const defaultState = {
   hiddenColumnIds: ["candidacyStatus", "notes"],
   filters: {
     applicantFilter: {
+      publishingGroups: [PublishingGroup.ItJobs, PublishingGroup.ItJobsOngoing],
```
Comment (vd1992) [inline apps/web/src/components/PoolCandidatesTable/PoolCandidatesTable.tsx:346]: I'm not 100% sold on this part
When you load up an IT pool like the CMO Digital Careers test pool this default filter changes nothing
But when you load up something else, like the IAP pool, you need to clear the default publishing group every time to have results

image

File: api/app/GraphQL/Queries/CountPoolCandidatesByPool.php
Comment (vd1992) [inline api/app/GraphQL/Queries/CountPoolCandidatesByPool.php:1]: Given these filters I get this on the admin side, and below it the no filters selected results on Find talent
Shouldn't they be equal or did I miss a filter somewhere? 

image

image

File: api/graphql/schema.graphql [hunk 2]
```diff
@@ -574,6 +574,7 @@ type ApplicantFilter {
   qualifiedClassifications: [Classification] @belongsToMany # Filters applicants based on the classifications pools they've qualified in.
   qualifiedStreams: [PoolStream] @rename(attribute: "qualified_streams") # Filters applicants based on the streams of the pools they've qualified in.
   pools: [Pool] @belongsToMany @can(ability: "view", resolved: true)
+  publishingGroups: [PublishingGroup] @rename(attribute: "publishing_groups")
```
[the remark this answers is not in the document]
Comment (esizer) [reply api/graphql/schema.graphql:577]: Oh, it looks like I actually put this field in the wrong spot 😓 Fixed in [update where new field is placed in schema](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7389/commits/4c859e5f74b517150d2b451059d86bbfd342990a)

File: apps/web/src/components/PoolCandidatesTable/PoolCandidatesTable.tsx (hunk 1 above)
Comment (esizer) [reply apps/web/src/components/PoolCandidatesTable/PoolCandidatesTable.tsx]: Yeah, good call. I'll ask about this in standup today 👌

File: api/app/GraphQL/Queries/CountPoolCandidatesByPool.php
Comment (esizer) [reply api/app/GraphQL/Queries/CountPoolCandidatesByPool.php:1]: Hmm, I'm not certain about this. I'm having a hard time reproducing it. Though, I forgot to test before making changes in [update where new field is placed in schema](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7389/commits/4c859e5f74b517150d2b451059d86bbfd342990a).

Screenshot 2023-08-02 084555

**Review by tristan-orourke — CHANGES_REQUESTED** [review]
Just remove publishingGroups field from ApplicantFilter in the schema, but besides that it looks good!

File: api/graphql/schema.graphql (hunk 2 above)
[the remark this answers is not in the document]
Comment (esizer) [reply api/graphql/schema.graphql]: Ahh yes, you are correct! Removed in [remove publishing groups from applicant filter](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7389/commits/bd954750cfdcf015cde88847776bf316fa565b33)

**Review by tristan-orourke — APPROVED** [review]

**Pull request merged**